crates/bevy_winit/src/system.rs

```diff
@@ -327,6 +330,17 @@ pub(crate) fn changed_windows(
                     winit_window.set_fullscreen(new_mode);
                 }
             }
+        } else {
+            #[cfg(target_os = "macos")]
+            match (winit_window.fullscreen(), window.mode) {
+                (Some(winit::window::Fullscreen::Borderless(_)), WindowMode::Windowed) => {
+                    window.mode = WindowMode::BorderlessFullscreen(MonitorSelection::Current);
```

Comment:
We should probably make sure we are copying the correct monitor returned from `winit` rather than just defaulting to `Current`.